lib/wasi/src/fs/fd.rs

```diff
@@ -74,6 +70,21 @@ impl InodeVal {
     }
 }
 
+#[derive(Debug)]
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+pub struct NotificationInner {
+    /// Used for event notifications by the user application or operating system
+    /// (positive number means there are events waiting to be processed)
+    pub counter: AtomicU64,
```

Comment:
None of this should be public, we should encapsulate all the logic in methods.

Reply:
fixed in https://github.com/wasmerio/wasmer/commit/4f117a6b63222d403a1e1bed03ff5fb15b3527ed